File: src/main/java/xyz/oribuin/eternalclaims/EternalClaims.java
package xyz.oribuin.eternalclaims;

import dev.rosewood.rosegarden.RosePlugin;
import dev.rosewood.rosegarden.manager.Manager;
import org.bukkit.plugin.PluginManager;
import xyz.oribuin.eternalclaims.hook.PAPI;
import xyz.oribuin.eternalclaims.listener.ClaimListeners;
import xyz.oribuin.eternalclaims.manager.ClaimManager;
import xyz.oribuin.eternalclaims.manager.CommandManager;
import xyz.oribuin.eternalclaims.manager.ConfigurationManager;
import xyz.oribuin.eternalclaims.manager.LocaleManager;

import java.util.List;

public class EternalClaims extends RosePlugin {

    private static EternalClaims instance;

    public EternalClaims() {
        super(-1, -1, ConfigurationManager.class, null, LocaleManager.class, CommandManager.class);

        instance = this;
    }

    public static EternalClaims getInstance() {
        return instance;
    }

    @Override
    protected void enable() {

        // Register PlaceholderAPI
        if (this.getServer().getPluginManager().getPlugin("PlaceholderAPI") != null) {
            new PAPI(this).register();
        }

        // Register Listeners
        PluginManager pluginManager = this.getServer().getPluginManager();
        pluginManager.registerEvents(new ClaimListeners(this), this);

    }

    @Override
    protected void disable() {

    }

    @Override
    protected List<Class<? extends Manager>> getManagerLoadPriority() {
        return List.of(ClaimManager.class);
    }

}

File: src/main/kotlin/dev/oribuin/eternalclaims/EternalClaims.kt
package dev.oribuin.eternalclaims

import dev.oribuin.eternalclaims.config.Setting
import dev.oribuin.eternalclaims.hook.PapiProvider
import dev.oribuin.eternalclaims.listener.ClaimListeners
import dev.oribuin.eternalclaims.manager.ClaimManager
import dev.oribuin.eternalclaims.manager.CommandManager
import dev.oribuin.eternalclaims.manager.LocaleManager
import dev.rosewood.rosegarden.RosePlugin
import dev.rosewood.rosegarden.config.RoseSetting
import dev.rosewood.rosegarden.manager.Manager

class EternalClaims : RosePlugin(
    -1,
    -1,
    null,
    LocaleManager::class.java,
    CommandManager::class.java
) {


    override fun enable() {
        // Register PlaceholderAPI
        if (server.pluginManager.getPlugin("PlaceholderAPI") != null) {
            PapiProvider(this).register()
        }

        // Register Listeners
        val pluginManager = server.pluginManager
        pluginManager.registerEvents(ClaimListeners(this), this)
    }

    override fun disable() {
    }

    override fun getManagerLoadPriority(): List<Class<out Manager>> {
        return listOf(ClaimManager::class.java)
    }

    override fun getRoseConfigSettings(): MutableList<RoseSetting<*>> {
        return Setting.keys
    }

    init {
        instance = this
    }

    companion object {
        lateinit var instance: EternalClaims
    }

}
